pkg/webhook/build.go

```diff
@@ -237,15 +237,14 @@ func validationError(reason, format string, fmtArgs ...interface{}) error {
 	}
 }
 
-func validateTimeout(timeout string) error {
+func validateTimeout(timeout time.Duration) error {
 	maxTimeout := time.Duration(24 * time.Hour)
 
-	if timeout == "" {
+	if timeout == 0 {
 		return nil
 	} else {
-		timeout, err := time.ParseDuration(timeout)
-		if err != nil {
-			return validationError("InvalidTimeFormat", "invalid build timeout %q err %#v. Refer https://golang.org/pkg/time/#ParseDuration for time format documentation", timeout, err)
+		if timeout < 0 {
+			return validationError("InvalidTimeFormat", "invalid build timeout %q. Refer https://golang.org/pkg/time/#ParseDuration for time format documentation", timeout)
 		}
 
 		// time out should not greater than 24 hours
```

Comment:
GitHub apparently won't let me comment on lines that weren't changed in this PR :unamused: but I think we don't need `Hours` on the line below this. Just `timeout > maxTimeout` has the same effect.